opentelemetry/src/test/scala/zio/telemetry/opentelemetry/TracingTest.scala

```diff
@@ -0,0 +1,197 @@
+package zio.telemetry.opentelemetry
+
+import io.opentelemetry.common.AttributeValue
+import io.opentelemetry.context.propagation.HttpTextFormat.{ Getter, Setter }
+import io.opentelemetry.exporters.inmemory.InMemoryTracing
+import io.opentelemetry.sdk.trace.TracerSdkProvider
+import io.opentelemetry.sdk.trace.data.SpanData
+import io.opentelemetry.trace.{ SpanId, Tracer }
+import zio.clock.Clock
+import zio.duration._
+import zio.telemetry.opentelemetry.Tracing.inject
+import zio.telemetry.opentelemetry.TracingSyntax._
+import zio.telemetry.opentelemetry.attributevalue.AttributeValueConverterInstances._
+import zio.test.Assertion._
+import zio.test.environment.TestClock
+import zio.test.{ assert, suite, testM, DefaultRunnableSpec }
+import zio.{ Has, UIO, URLayer, ZIO, ZLayer }
+
+import scala.collection.mutable
+import scala.jdk.CollectionConverters._
+
+object TracingTest extends DefaultRunnableSpec {
+
+  def createMockTracer: (InMemoryTracing, Tracer) = {
+    val tracerProvider: TracerSdkProvider = TracerSdkProvider.builder().build()
+    val inMemoryTracing                   = InMemoryTracing.builder().setTracerProvider(tracerProvider).build()
+    val tracer                            = tracerProvider.get("InMemoryTracing")
+
+    (inMemoryTracing, tracer)
+  }
+
+  def getFinishedSpans(inMemoryTracing: InMemoryTracing): List[SpanData] =
+    inMemoryTracing.getSpanExporter.getFinishedSpanItems.asScala.toList
+
+  def createMockLayer: URLayer[Clock, Has[InMemoryTracing] with Tracing] = {
+    val (inMemoryTracing, tracer) = createMockTracer
+    ZLayer.succeed(inMemoryTracing) ++ Tracing.live(tracer)
+  }
+
+  def spec =
+    suite("zio opentelemetry")(
+      testM("acquire/release the service") {
+        val (inMemoryTracing, tracer) = createMockTracer
+        lazy val finishedSpans        = getFinishedSpans(inMemoryTracing)
+
+        Tracing
+          .live(tracer)
+          .build
+          .use_(UIO.unit)
+          .as(assert(finishedSpans)(hasSize(equalTo(0))))
+      },
+      suite("spans")(
+        testM("childSpan") {
+          for {
+            inMemoryTracing <- ZIO.access[Has[InMemoryTracing]](_.get)
+            _               <- UIO.unit.span("Child").span("Root")
+            spans           = getFinishedSpans(inMemoryTracing)
+            root            = spans.find(_.getName == "Root")
+            child           = spans.find(_.getName == "Child")
+          } yield assert(root)(isSome(anything)) &&
+            assert(child)(
+              isSome(
+                hasField[SpanData, SpanId](
+                  "parentSpanId",
+                  _.getParentSpanId,
+                  equalTo(root.get.getSpanId)
+                )
+              )
+            )
+        }.provideCustomLayer(createMockLayer),
```

Comment:
`.provideCustomLayer(createMockLayer)` could be moved to the `suite` level in order to save a few repetitions.

Reply:
Fixed with https://github.com/zio/zio-telemetry/pull/119/commits/3508773a746bb0c3d5d6722294ccdbea981e601d